rwengine/src/ai/CharacterController.cpp

```diff
@@ -239,6 +239,28 @@ bool Activities::EnterVehicle::update(CharacterObject *character,
         } else if (character->animator->getAnimation(AnimIndexAction) ==
                    anm_enter) {
             if (character->animator->isCompleted(AnimIndexAction)) {
+                if (vehicle->getVehicle()->vehiclename_ == "TAXI"
+                    || vehicle->getVehicle()->vehiclename_ == "CABBIE"
+                    || vehicle->getVehicle()->vehiclename_ == "BORGNINE") {
+                    // Earn $25 from taxi cab
+                    // @todo
+                } else if (vehicle->getVehicle()->vehiclename_ == "POLICAR") {
+                    // Police cruisers give 5 shotgun cartridges
+                    character->addToInventory(4, 5);
```

Comment:
That's a tricky part. I've searched quite a bit, couldn't find any snippets in the code yet. I thought of making sure the character's model is Claude, that perhaps would work better than matching the XYZ coords of the player model against the character object.

As for the original game -- I've been giving rides to NPCs in police cruisers during missions, they never get the shotgun. I don't recall witnessing an  NPC stealing a special vehicle ever. Thieves only steal fancy cars, so neither taxis nor cruisers/ambulances can be initially boarded by non-player characters inside the game, at least to my knowledge.
I have not seen a case where a player would enter a special vehicle as a non-driver as well. It is likely that the "bonus" gets depleted once the vehicle gets boarded by the player character for the first time.

If you could give me a creative hint at how to test it, aside from REing the thing (which I'm honestly, below horrible with), I'd really appreciate it.